tests/unit/cli/test_pipeline.py

```diff
@@ -11,3 +11,238 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
 #  or implied. See the License for the specific language governing
 #  permissions and limitations under the License.
+
+import os
+import subprocess
+
+from click.testing import CliRunner
+
+from zenml.cli.pipeline import pipeline
+from zenml.enums import ExecutionStatus
+from zenml.repository import Repository
+from zenml.utils import yaml_utils
+
+PIPELINE_NAME = "some_pipe"
+STEP_NAME = "some_step"
+MATERIALIZER_NAME = "SomeMaterializer"
+CUSTOM_OBJ_NAME = "SomeObj"
+
+object_definition = f"""
```

Comment:
Maybe as a general point on this approach: Wouldn't it be easier to just have python files inside our tests which we can copy into the clean repo and try to run them? I think it would make the tests a little easier to understand, and at the same time allow easier refactoring and detecting of errors in those files